src/main/java/seedu/address/model/session/RecurringSession.java

```diff
@@ -75,7 +75,9 @@ private boolean startAfter(SessionDate sessionDate) {
      */
     public boolean hasSessionOnDate(SessionDate sessionDate) {
         requireAllNonNull(sessionDate);
-        return !endBefore(sessionDate) && isConsistentDatesAndInterval(getSessionDate(), sessionDate, getInterval());
+        return !startAfter(sessionDate)
```

Comment:
hey @nowknowing, was the function outputting incorrect results prior to this?

Reply:
no it wasn't because isConsistentDatesAndInterval does the calculation as well

Reply:
but i guess its simpler (less computation) if use startSafter(sessionDate) and more consistent

Reply:
ah, you're referring to the case where it'll break the condition immediately if it doesn't meet `!startAfter`?